### PR TITLE
add dynamic_embedding.Variable clear

### PR DESCRIPTION
**Pull request by yuanqingsunny**
# Description

Brief Description of the PR:

 New Feature: add dynamic_embedding.Variable clear op

### CODE REVIEW

**Comment by yuanqingsunny** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Ftensorflow%2Frecommenders-addons%2Fpull%2F100) for more info**.

@googlebot I signed it!

**Comment by yuanqingsunny** [comment]
> @googlebot I fixed it.

@googlebot I fixed it.

**Comment by yuanqingsunny** [comment]
> We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for all the commit author(s) or Co-authors. If you authored these, maybe you used a different email address in the git commits than was used to sign the CLA ([login here](https://cla.developers.google.com/) to double check)? If these were authored by someone else, then they will need to sign a CLA as well, and confirm that they're okay with these being contributed to Google.
> In order to pass this check, please resolve this problem and then comment `@googlebot I fixed it.`. If the bot doesn't comment, it means it doesn't think anything has changed.
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Ftensorflow%2Frecommenders-addons%2Fpull%2F100) for more info**.

@googlebot I fixed it.

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op.cc [hunk 1]
```diff
@@ -365,6 +370,34 @@ class HashTableRemoveOp : public HashTableOpKernel {
   }
 };
 
+REGISTER_KERNEL_BUILDER(Name("TFRA>CuckooHashTableRemove").Device(DEVICE_CPU),
```
Comment (rhdong) [inline tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op.cc:373]: Pls sue `PREFIX_OP_NAME(CuckooHashTableRemove)`

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc [hunk 2]
```diff
@@ -15,6 +15,7 @@ limitations under the License.
 #if GOOGLE_CUDA
 
 #include "tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.h"
+
```
Comment (rhdong) [inline tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc:18]: Extra blank lines

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc [hunk 3]
```diff
@@ -483,15 +513,21 @@ REGISTER_KERNEL_BUILDER(
...
+#define REGISTER_KERNEL(key_dtype, value_dtype)                            \
+  REGISTER_KERNEL_BUILDER(                                                 \
+      Name(PREFIX_OP_NAME(CuckooHashTableOfTensors))                       \
+          .Device(DEVICE_GPU)                                              \
+          .TypeConstraint<key_dtype>("key_dtype")                          \
+          .TypeConstraint<value_dtype>("value_dtype"),                     \
+      HashTableGpuOp<                                                      \
+          lookup::CuckooHashTableOfTensorsGpu<key_dtype, value_dtype>,     \
+          key_dtype, value_dtype>);                                        \
+  REGISTER_KERNEL_BUILDER(Name(PREFIX_OP_NAME(CuckooHashTableClear))       \
```
Comment (rhdong) [inline tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc:526]: It's better to add this code block below [`HashTableClearGpuOp` class implement](https://github.com/yuanqingsunny/recommenders-addons/blob/f2fd10dad6f213a41c023e465b9cdea9c0b12e48/tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc#L440).

File: tensorflow_recommenders_addons/dynamic_embedding/core/ops/cuckoo_hashtable_ops.cc [hunk 4]
```diff
@@ -181,6 +181,19 @@ REGISTER_OP(PREFIX_OP_NAME(CuckooHashTableRemove))
...
+
+REGISTER_OP(PREFIX_OP_NAME(CuckooHashTableClear))
+    .Input("table_handle: resource")
+    .Attr("key_dtype: type")
+    .Attr("value_dtype: type")
+    .SetShapeFn([](InferenceContext* c) {
+      ShapeHandle handle;
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 0, &handle));
+
+      // TODO(turboale): Validate keys shape.
```
Comment (rhdong) [inline tensorflow_recommenders_addons/dynamic_embedding/core/ops/cuckoo_hashtable_ops.cc:193]: remove it

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op.cc (hunk 1 above)
Comment (yuanqingsunny) [reply tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op.cc]: fix

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc (hunk 2 above)
Comment (yuanqingsunny) [reply tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc:18]: fix

File: tensorflow_recommenders_addons/dynamic_embedding/core/ops/cuckoo_hashtable_ops.cc (hunk 4 above)
Comment (yuanqingsunny) [reply tensorflow_recommenders_addons/dynamic_embedding/core/ops/cuckoo_hashtable_ops.cc]: fixed

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc (hunk 3 above)
Comment (yuanqingsunny) [reply tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc:525]: put here to keep same key_dtype and value_dtype with CuckooHashTableOfTensors

**Review by rhdong — APPROVED** [review]
LGTM

**Pull request merged**